examples/tutorials/masterfiles_policy_framework_upgrade.markdown

```diff
@@ -171,724 +138,798 @@ Changes not staged for commit:
-```./configure``` and ```make install```
-),
-[installed masterfiles tarball](https://cfengine-package-repos.s3.amazonaws.com/tarballs/cfengine-masterfiles-{{site.cfengine.branch}}.{{site.cfengine.latest_patch_release}}.pkg.tar.gz) (ready
-for extraction),
-or
-[directly from github](https://github.com/cfengine/masterfiles/tree/{{site.cfengine.masterfiles_branch}}).
 
-We will install the MPF from source obtained
-directly from github.
+### Installing from Git
 
-**Note:** You will need ```automake``` to install from source.
+First, clone the desired version of the MPF source.
 
-First clone the masterfiles repository for the version you are installing. And
-verify you have the correct tag checked out.
-
-**Note:** Directly checking out a tag as in the example below is only
-supported in Git versions 1.7.9.5 and newer.
+```bash
+exec 2>&1
+MPF_VERSION="3.18.5"
+git clone -b $MPF_VERSION https://github.com/cfengine/masterfiles $INTEGRATION_ROOT/masterfiles-source-$MPF_SEED_VERSION
+:
+```
 
-```console
-[root@hub MPF_upgrade]# cd ..
-[root@hub ~]# git clone -b 3.7.4 https://github.com/cfengine/masterfiles
-[root@hub ~]# cd masterfiles
-[root@hub ~]# git describe
-3.7.4
 ```
+Cloning into '/tmp/MPF-upgrade/integration/masterfiles-source-'...
+Note: switching to 'ea81b7b790f1bf98cbf145d8700a06b37d74fb84'.
 
-**Note:** For systems without python 3 easily available (such as centos 6) you can use the following to get around problems in autogen with 3rdparty/core/determine-version.py, which requires python3.
+You are in 'detached HEAD' state. You can look around, make experimental
+changes and commit them, and you can discard any commits you make in this
+state without impacting any branches by switching back to a branch.
+
+If you want to create a new branch to retain commits you create, you may
+do so (now or later) by using -c with the switch command. Example:
+
+  git switch -c <new-branch-name>
+
+Or undo this operation with:
+
+  git switch -
+
+Turn off this advice by setting config variable advice.detachedHead to false
 
-```
-export EXPLICIT_VERSION=$(git describe)
 ```
 
-Now we will install the masterfiles from upstream into the directory where we
-are doing the integration.
+Then build and install targeting the integration root directory. When installed from source masterfiles installs into the `masterfiles` directory.
 
-First we build and install masterfiles to a temporary location.
+```bash
+exec 2>&1
+cd $INTEGRATION_ROOT/masterfiles-source-$MPF_VERSION
+export EXPLICIT_VERSION=$MPF_VERSION
 
-```console
 ./autogen.sh
-[root@hub masterfiles]# ./autogen.sh
-configure.ac:31: installing `./config.guess'
-configure.ac:31: installing `./config.sub'
-configure.ac:34: installing `./install-sh'
-configure.ac:34: installing `./missing'
-checking build system type... x86_64-unknown-linux-gnu
-checking host system type... x86_64-unknown-linux-gnu
-checking target system type... x86_64-unknown-linux-gnu
+make
+make install prefix=$INTEGRATION_ROOT/
```

Comment:
DESTDIR here?

Reply:
I know, I oped against it becasue it makes a deep directory, and for someone who is probably cloning the root of their policy to a place it's a bit weird to me to have them be sure to clone it under var/cfengine

Reply:
DESTDIR makes a deeper copy with many subdirs like `var/cfengine` so awkward for this procedure.